diff --git a/meson.build b/meson.build
index 69c1bb8..fc6302b 100644
--- a/meson.build
+++ b/meson.build
@@ -81,6 +81,7 @@ prog_python = import('python').find_installation('python3')
 
 with_host_windows = host_machine.system() == 'windows'
 with_host_darwin = host_machine.system() == 'darwin'
+with_host_android = cc.has_define('__ANDROID__')
 
 not_found = dependency('', required: false)
 gbm_dep = not_found
@@ -295,7 +296,7 @@ have_glx = false
 
 if with_egl
    if cc.has_header('epoxy/egl.h', dependencies: epoxy_dep) and epoxy_dep.get_variable(pkgconfig: 'epoxy_has_egl') == '1'
-      if with_host_windows or with_host_darwin
+      if with_host_windows or with_host_darwin or with_host_android
          have_egl = true
       else
          if libdrm_dep.found()
